src/server/req_quejob.c

```diff
@@ -261,6 +263,10 @@ validate_perm_res_in_select(char *val)
 							return PBSE_SYSTEM;
 						return PBSE_INVALSELECTRESC;
 					}
+				} else if (val_exist) {
+					if ((resc_in_err = strdup(pkvp[j].kv_keyw)) == NULL)
```

Comment:
Add a comment here that 'resc_in_err' would get freed if function returns PBSE_UNKRESC, and the function call is followed by a req_reject() which would cause set_err_msg() to be called, which in turn would free(resc_in_err). Need to add this comment so as to not have another developer to inadvertently set the return value to something else, causing a memory leak. Actually in src/server/reply_send.c:set_err_msg(), there are more return code checked before doing a free():
#ifndef PBS_MOM
        } else if ((
                (code == PBSE_UNKRESC)         ||
                (code == PBSE_BADATVAL)        ||
                (code == PBSE_INVALSELECTRESC) ||
                (code == PBSE_INVALJOBRESC)    ||
                (code == PBSE_INVALNODEPLACE)  ||
                (code == PBSE_DUPRESC)         ||
                (code == PBSE_INDIRECTHOP)     ||
                (code == PBSE_SAVE_ERR)        ||
                (code == PBSE_INDIRECTBT))
                && (resc_in_err != NULL)) {
                strncpy(msgbuf, pbse_to_txt(code), msglen-2);
                /* -2 is to make sure there is room for a colon and a space */
                strcat(msgbuf, ": ");
                strncat(msgbuf, resc_in_err, msglen - strlen(msgbuf));
                free(resc_in_err);
                resc_in_err = NULL;